repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -99,6 +102,48 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
   def this(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode) =
     this(originalResponse, headerMode, bodyMode, originalResponse.getOutputStream)
 
+  override def setStatus(i: Int, s: String): Unit = {
+    message = s
+    super.setStatus(i, s)
```

Comment:
Internally the way the `setStatus(i, s)` variant is used seems to be wrong in several places. The `String` parameter is what the `ResponseMessageService` is keying off of. This looks like a bastardization of a deprecated call to get the desired behavior. When I ran across it a year or so ago, we didn't have time to fix it and we still haven't got back around to clean it up yet.